packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -315,9 +335,11 @@ const _BaseButton: React.ForwardRefRenderFunction<BladeElementRef, BaseButtonPro
   },
   ref,
 ) => {
+  const isLink = Boolean(href);
   const childrenString = getStringFromReactText(children);
   const buttonRef = useBladeInnerRef(ref);
-  const disabled = isLoading || isDisabled;
+  // Button cannot be disabled when its rendered as Link
+  const disabled = isLoading || (isDisabled && !isLink);
```

Comment:
why so? if a link can be disabled then why not a button as a link? imagine an icon button - a particular user is not authorised to click it(regardless of a `button` or `a`) so link shouldn't be treated as special case?

Reply:
Link can't be disabled. Only `<Link variant="button" />` can be disabled.

If you go here, and change to `variant="anchor"` in story controls, it will end up getting enabled. https://blade.razorpay.com/?path=/story/components-link--disabled-link-button&globals=measureEnabled:false

In HTML itself, anchor tag can't be disabled.